core/src/main/scala/fs2/io/ssh/MinaFuture.scala

```diff
@@ -121,9 +114,7 @@ private[ssh] object MinaFuture {
     def exception(s: IoReadFuture) = s.getException()
 
     def result(s: IoReadFuture) = {
-      val buffer = s.getBuffer()
-      val length = s.getRead()
-      Chunk.bytes(buffer.array(), 0, length)
+      Chunk.array(s.getBuffer().array())
```

Comment:
This should be reverted to the previous impl, but using `Chunk.array(Array[Byte], Int, Int)` instead.

The problem with the current implementation is that `Buffer#array` returns the raw underlying `Array[Byte]`, which may be physically larger than the number of bytes read. `IoReadFuture#getRead` tells us how many bytes were actually read, so we know how much of the array it is safe to consume.

Reply:
Oh good to know! @gingerbreadmanphoto could you document this?

Reply:
Yeah 👍